src/ajax.js

```diff
@@ -54,8 +54,11 @@ export function ajax(url, callback, data, options = {}) {
     if (options.withCredentials) {
       x.withCredentials = true;
     } else {
-      x.setRequestHeader('X-Requested-With', 'XMLHttpRequest');
-      x.setRequestHeader('Content-Type', 'application/json;charset=UTF-8');
+      if (options.preflight !== false) {
```

Comment:
Any reason for the double negation here? Also don't see why we can't use `withCredentials` with `preflight`. IMO these should all be standalone options